src/components/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -0,0 +1,141 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { equals, reverse, slice } from 'ramda';
+import styled from 'styled-components';
+import { isNotUndefined } from 'ramda-adjunct';
+
+import { BreadcrumbsProps } from './Breadcrumbs.types';
+import { ColorTypes, Inline, SpaceSizes, getFontSize } from '../..';
+import { SSCIconNames } from '../../theme/icons/icons.enums';
+import { Icon } from '../Icon';
+import BreadcrumbItem from './BreadcrumbItem';
+import { DropdownMenu } from '../_internal/BaseDropdownMenu';
+import { ActionKinds } from '../../types/action.types';
+
+const IconWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  font-size: ${getFontSize('sm')};
+  width: 1rem;
+  height: 1rem;
+`;
+
+const BreadcrumbsWrapper = styled.div`
+  display: flex;
+`;
+
+// Build list of breadcrumbs interspersing a separator
+const insertSeparators = (items: React.ReactElement[]) => {
+  return items.reduce((prev, current, index) => {
+    let t = prev;
+    if (index < items.length - 1) {
+      t = t.concat(
+        current,
+        // eslint-disable-next-line react/no-array-index-key
+        <IconWrapper key={`separator-${index}`} aria-hidden>
+          <Icon
+            color={ColorTypes.neutral600}
+            name={SSCIconNames.chevronRight}
+          />
+        </IconWrapper>,
+      );
+    } else {
+      t.push(current);
+    }
+
+    return t;
+  }, []);
+};
+
+const Breadcrumbs: React.FC<BreadcrumbsProps> = ({ children, ...props }) => {
+  const itemsAfterCollapse = 2;
+  const itemsBeforeCollapse = 1;
+  const maxItems = 3;
+
+  // this renders the list of items only when the quantity
+  // is bigger than maxItems
+  const renderItemsBeforeAndAfter = (
+    allItems: React.ReactElement[],
+    allDropdownActions: ActionKinds<React.MouseEvent[]>[],
+  ) => {
+    const dropdown = () => {
+      return (
+        <DropdownMenu
+          key="breadcrumbs-dropdown"
+          actions={allDropdownActions}
+          paneWidth={270}
+          placement="bottom-start"
+          aria-hidden
+        >
+          <BreadcrumbItem>
+            <IconWrapper>
+              <Icon
+                aria-label="Open breadcrumbs menu"
+                name={SSCIconNames.ellipsisH}
+              />
+            </IconWrapper>
+          </BreadcrumbItem>
+        </DropdownMenu>
+      );
+    };
+
+    return [
+      ...slice(0, itemsBeforeCollapse, allItems),
+      dropdown(),
+      ...slice(allItems.length - itemsAfterCollapse, allItems.length, allItems),
+    ];
+  };
```

Comment:
Can we move this outside of `Breadcrumbs` component it doesn't seem like something that needs to be reinitializes on every render